File: leetcode/1343_Number_of_Sub-arrays_of_Size_K_and_Average_Greater_than_or_Equal_to_Threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int n=arr.length;
        int sum=0;
        for(int i=0;i<k;i++){
            sum+=arr[i];
        }
        int cnt=0;
        if(sum/k>=threshold)cnt++;
        for(int i=k;i<n;i++){
            sum+=arr[i];
            sum-=arr[i-k];
            if(sum/k>=threshold)cnt++;
        }
        return cnt;
    }
}
public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] arr = {2, 1, 3, 4, 1};
        int k = 3;
        int threshold = 2;
        int result = solution.numOfSubarrays(arr, k, threshold);
        System.out.println(result); // Output: 3
    }
}   

File: leetcode/1343_Number_of_Sub-arrays_of_Size_K_and_Average_Greater_than_or_Equal_to_Threshold.cpp
#include<bits/stdc++.h>
using namespace std;
    class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int n=arr.size();
        int sum=0;
        for(int i=0;i<k;i++){
            sum+=arr[i];
        }
        int cnt=(sum/k>=threshold);
        for(int i=k;i<n;i++){
            sum+=arr[i];
            sum-=arr[i-k];
            if(sum/k>=threshold)cnt++;
        }
        return cnt;
    }
};
int main(){
    Solution s;
    vector<int> arr={2,2,2,2,5,5,5,8};
    int k=3;
    int threshold=4;
    int ans=s.numOfSubarrays(arr,k,threshold);
    cout<<ans;
    return 0;
}